### PR TITLE
Add a section on number conversion

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Like https://github.com/mdn/content/pull/20234, but for numbers.

### Motivation

We use handwavy statements for number conversion in a lot of places. Time to formally introduce it.

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/number/index.md [hunk 1]
```diff
@@ -58,6 +58,45 @@ Integers can only be represented without loss of precision in the range -2<sup>5
 
 More details on this are described in the [ECMAScript standard](https://tc39.es/ecma262/#sec-ecmascript-language-types-number-type).
 
+### Number coercion
+
+Many built-in operations that expect numbers would first coerce their arguments to numbers (which is largely why `Number` objects behave similarly to number primitives). [The operation](https://tc39.es/ecma262/#sec-tonumber) can be summarized as follows:
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/number/index.md:63]:
```suggestion
Many built-in operations that expect numbers first coerce their arguments to numbers (which is largely why `Number` objects behave similarly to number primitives). [The operation](https://tc39.es/ecma262/#sec-tonumber) can be summarized as follows:
```

File: files/en-us/web/javascript/reference/operators/addition/index.md [hunk 2]
```diff
@@ -24,7 +24,11 @@ x + y
 
 ## Description
 
-The addition operator (`+`) is overloaded for two distinct operations: numeric addition and string concatenation. When evaluating, it would first coerce both operands to primitives by calling the object's [`[@@toPrimitive]()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive) (with `"default"` as hint), `valueOf()`, and `toString()` methods, in that order. Then, if one side is a string, the other operand is also [converted to a string](/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_coercion) and concatenated. Otherwise, both sides are converted to numbers (or, if they are [BigInts](/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt), stay as BigInts), and numeric addition is performed.
+The addition operator (`+`) is overloaded for two distinct operations: numeric addition and string concatenation. When evaluating, it would first coerce both operands to primitives by calling the object's [`[@@toPrimitive]()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive) (with `"default"` as hint), `valueOf()`, and `toString()` methods, in that order. Then, the two operands' types are tested:
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/operators/addition/index.md:27]:
```suggestion
The addition operator (`+`) is overloaded for two distinct operations: numeric addition and string concatenation. When evaluating, it first coerces both operands to primitives by calling the object's [`[@@toPrimitive]()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive) (with `"default"` as hint), `valueOf()`, and `toString()` methods, in that order. Then, the two operands' types are tested:
```

File: files/en-us/web/javascript/reference/operators/equality/index.md [hunk 3]
```diff
@@ -40,8 +40,8 @@ The equality operators (`==` and `!=`) provide the [IsLooselyEqual](/en-US/docs/
...
    - If they are of the same type, compare them using step 1.
    - If one of the operands is a Symbol but the other is not, return `false`.
-   - If one of the operands is a Boolean but the other is not, convert the boolean to a number: `true` is converted to 1, and `false` is converted to 0. Then compare the two operands loosely again.
-   - Number to String: convert the string to a Number using the same algorithm as the [`Number()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/Number) constructor. Conversion failure would result in `NaN`, which will guarantee the equality to be `false`.
+   - If one of the operands is a Boolean but the other is not, [convert the boolean to a number](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion): `true` is converted to 1, and `false` is converted to 0. Then compare the two operands loosely again.
+   - Number to String: [convert the string to a number](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion). Conversion failure would result in `NaN`, which will guarantee the equality to be `false`.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/operators/equality/index.md:44]:
```suggestion
   - Number to String: [convert the string to a number](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion). Conversion failure result in `NaN`, which will guarantee the equality to be `false`.
```

File: files/en-us/web/javascript/equality_comparisons_and_sameness/index.md [hunk 4]
```diff
@@ -85,8 +85,8 @@ Loose equality is _symmetric_: `A == B` always has identical semantics to `B ==
...
    - If they are of the same type, compare them using step 1.
    - If one of the operands is a Symbol but the other is not, return `false`.
-   - If one of the operands is a Boolean but the other is not, convert the boolean to a number: `true` is converted to 1, and `false` is converted to 0. Then compare the two operands loosely again.
-   - Number to String: convert the string to a Number using the same algorithm as the [`Number()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/Number) constructor. Conversion failure would result in `NaN`, which will guarantee the equality to be `false`.
+   - If one of the operands is a Boolean but the other is not, [convert the boolean to a number](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion): `true` is converted to 1, and `false` is converted to 0. Then compare the two operands loosely again.
+   - Number to String: [convert the string to a number](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion). Conversion failure would result in `NaN`, which will guarantee the equality to be `false`.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/equality_comparisons_and_sameness/index.md:89]:
```suggestion
   - Number to String: [convert the string to a number](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion). Conversion failure results in `NaN`, which will guarantee the equality to be `false`.
```

File: files/en-us/web/javascript/reference/operators/addition/index.md [hunk 5]
```diff
@@ -24,7 +24,11 @@ x + y
...
+- Otherwise, both sides are [converted to numbers](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_coercion), and numeric addition is performed.
 
 String concatenation is often thought to be equivalent with [template literals](/en-US/docs/Web/JavaScript/Reference/Template_literals) or [`String.prototype.concat()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat), but they are not. Addition would coerce the expression to a _primitive_, which calls [`valueOf()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/valueOf) in priority; on the other hand, template literals and `concat()` would coerce the expression to a _string_, which calls [`toString()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) in priority. If the expression has a [`@@toPrimitive`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive) method, string concatenation calls it with `"default"` as hint, while template literals use `"string"`. This is important for objects that have different string and primitive representations — such as [Temporal](https://github.com/tc39/proposal-temporal), whose `valueOf()` method throws.
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/operators/addition/index.md:33]:
```suggestion
String concatenation is often thought to be equivalent with [template literals](/en-US/docs/Web/JavaScript/Reference/Template_literals) or [`String.prototype.concat()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat), but they are not. Addition coerces the expression to a _primitive_, which calls [`valueOf()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/valueOf) in priority; on the other hand, template literals and `concat()` would coerce the expression to a _string_, which calls [`toString()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) in priority. If the expression has a [`@@toPrimitive`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive) method, string concatenation calls it with `"default"` as hint, while template literals use `"string"`. This is important for objects that have different string and primitive representations — such as [Temporal](https://github.com/tc39/proposal-temporal), whose `valueOf()` method throws.
```

**Review by sideshowbarker — APPROVED** [review]
Looks great to me, with just suggestions for dropping instances of “would”.

**Comment by Josh-Cena** [comment]
Thanks, they keep creeping back into my writing.

**Comment by sideshowbarker** [comment]
> Thanks, they keep creeping back into my writing.

Yeah I noticed in an earlier patch too and I guess I should have flagged it before, but it’s not outright wrong so I didn’t. But this patch just had a several of them, so I figured it was worth suggesting to zap them

**Pull request merged**